k2/csrc/rnnt_decode.h

```diff
@@ -0,0 +1,285 @@
+/**
+ * Copyright      2022  Xiaomi Corporation (authors:Daniel Povey, Wei kang)
+ *
+ * See LICENSE for clarification regarding multiple authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef K2_CSRC_RNNT_DECODE_H_
+#define K2_CSRC_RNNT_DECODE_H_
+
+#include <algorithm>
+#include <memory>
+#include <vector>
+
+#include "k2/csrc/array.h"
+#include "k2/csrc/array_of_ragged.h"
+#include "k2/csrc/array_ops.h"
+#include "k2/csrc/log.h"
+#include "k2/csrc/macros.h"
+
+namespace k2 {
+namespace rnnt_decoding {
+
+/*
+  The RNN-T decoding implemented here is for what we call "modified" RNN-T, or
+  equivalently, regular RNN-T but with max_sym_per_frame set to 1.  (We can
+  train with the "modified" option set with some probability, in order to ensure
+  that the trained model is compatible with decoding with this setting).
+
+  - contexts are finite symbol left-contexts, of length
+  RnntDecodingConfig::decoder_history_len. conceptuatly they represent a list of
+  `decoder_history_len` symbols; they are represented numerically as, for
+  example in the length-2-history case: symbol_{t-1} +  symbol_{t-2} *
+  num_symbols.
+
+  - frames come from the predictor network, they are derived from sub-sampling
```

Comment:
```suggestion
  - frames come from the transcription network, they are derived from sub-sampling
```

The transcription network is known as `encoder`, which processes acoustic information.

The predictor network is also named as `decoder`.